src/components/groups/PostList.tsx

```
@@ -162,20 +256,49 @@ export function PostList({ communityId, showOnlyApproved = false }: PostListProp
-    ? processedPosts.filter(post => 'approval' in post)
-    : processedPosts;
+  // Sort posts based on sortBy
+  const sortedPosts = [...filteredPosts].sort((a, b) => {
+    switch (sortBy) {
+      case "oldest":
+        return a.created_at - b.created_at;
+      case "popular":
+        return (b.reactions?.likes || 0) - (a.reactions?.likes || 0);
+      case "trending":
+        // Simple trending algorithm: (likes + comments * 2) / hours since post
+        const hoursA = (Date.now() / 1000 - a.created_at) / 3600;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)